pkg/reconciler/reconciler.go

```diff
@@ -429,6 +437,30 @@ func WithSelector(s metav1.LabelSelector) Option {
 	}
 }
 
+// WithControllerSetupFunc is an Option that allows customizing a controller before it is started.
+// The only supported customization here is adding additional Watch sources to the controller.
+func WithControllerSetupFunc(f ControllerSetupFunc) Option {
+	return func(r *Reconciler) error {
+		r.controllerSetupFuncs = append(r.controllerSetupFuncs, f)
+		return nil
+	}
+}
+
+// ControllerSetup allows restricted access to the Controller using the WithControllerSetupFunc option.
+// Currently the only supposed configuration is adding additional watchers do the controller.
+type ControllerSetup interface {
+	// Watch takes events provided by a Source and uses the EventHandler to
+	// enqueue reconcile.Requests in response to the events.
+	//
+	// Watch may be provided one or more Predicates to filter events before
+	// they are given to the EventHandler.  Events will be passed to the
+	// EventHandler if all provided Predicates evaluate to true.
+	Watch(src source.Source, eventhandler handler.EventHandler, predicates ...predicate.Predicate) error
```

Comment:
The intention here was to simply expose the controller with limited configuration capabilities.

I'm not sure how I would go about incorporating your suggestion given that `controller.Controller` provides 

```go
    Watch(src source.Source, eventhandler handler.EventHandler, predicates ...predicate.Predicate)
```

and not

```go
    Watch(src source.Source, eventhandler handler.EventHandler, opts ...builder.WatchesOption)
```

(which is implemented by `Builder`).

Essentially I would have to transform `WatchesOption`s to `Predicate`s. But given that the former is more general, there is no general way to do so with the current controller-runtime?

We have settled on exposing `Controller` as opposed to `Builder` since the `Builder` came with certain limitations that made it impossible to do what we required from it. `Controller` on the other hand is sufficient.